tools/wptrunner/wptrunner/executors/platformaccessibility/executoraxapi.py

```diff
@@ -0,0 +1,147 @@
+# mypy: allow-untyped-defs
+
+from ApplicationServices import (
+    AXUIElementCopyAttributeNames,
+    AXUIElementCopyAttributeValue,
+    AXUIElementCreateApplication,
+)
+
+from Cocoa import (
+    NSApplicationActivationPolicyRegular,
+    NSPredicate,
+    NSWorkspace,
+)
+
+def find_browser(name):
+    """Find the AXUIElement representing the browser.
+
+    :param name: The name of the browser.
+    :return: AXUIElement or None.
+    """
+
+    ws = NSWorkspace.sharedWorkspace()
+    regular_predicate = NSPredicate.predicateWithFormat_(
+        f"activationPolicy == {NSApplicationActivationPolicyRegular}"
+    )
+    running_apps = ws.runningApplications().filteredArrayUsingPredicate_(
+        regular_predicate
+    )
+    name_predicate = NSPredicate.predicateWithFormat_(
+        f"localizedName contains[c] '{name}'"
+    )
+    filtered_apps = running_apps.filteredArrayUsingPredicate_(name_predicate)
+    if filtered_apps.count() == 0:
+        return None
+    app = filtered_apps[0]
+    pid = app.processIdentifier()
+    if pid == -1:
+        return None
+    return AXUIElementCreateApplication(pid)
+
+
+def find_active_tab(browser):
+    """Find the active tab of the browser.
+
+    :param browser: The name of the browser.
+    :return: AXUIElement representing test document or None.
+    """
+
+    stack = [browser]
+    while stack:
+        node = stack.pop()
+
+        (err, role) = AXUIElementCopyAttributeValue(node, "AXRole", None)
+        if err:
+            continue
+        if role == "AXWebArea":
+            return node
+
+        (err, children) = AXUIElementCopyAttributeValue(node, "AXChildren", None)
+        if err:
+            continue
+        stack.extend(children)
+
+    return None
+
+
+def find_node(root, attribute, expected_value):
+    """Find the AXUIElement with a specified dom_id.
+
+    :dom_id: The dom ID.
+    :return: AXUIElement or None if not found.
+    """
+
+    stack = [root]
+    while stack:
+        node = stack.pop()
+
+        (err, attributes) = AXUIElementCopyAttributeNames(node, None)
+        if err:
+            continue
+        if attribute in attributes:
+            (err, value) = AXUIElementCopyAttributeValue(node, attribute, None)
+            if err:
+                continue
+            if value == expected_value:
+                return node
+
+        (err, children) = AXUIElementCopyAttributeValue(node, "AXChildren", None)
+        if err:
+            continue
+        stack.extend(children)
+    return None
+
+
+def serialize_node(node):
```

Comment:
I don't think this is called anywhere? What you could do is log the desired `node`s at the `debug` level, which filtered out by default for most formatters, but lets users opt-in without requiring a code change.

Same comment for `serialize_node` in `executorwindowsaccessibility.py`.

Reply:
Thanks for the suggestion! I'll use the logger.debug to print the serialized node. It is only for printing the node when debugging/writing tests/more test endpoints anyway.